graph/push_v2.go

```diff
@@ -114,13 +114,17 @@ func (p *v2Pusher) pushV2Tag(tag string) error {
 	}
 
 	out := p.config.OutStream
+	layersSeen := make(map[string]bool)
 
 	for ; layer != nil; layer, err = p.graph.GetParent(layer) {
 		if err != nil {
 			return err
 		}
 
-		if p.layersSeen[layer.ID] {
+		// break early if layer has already been seen in this image,
+		// this prevents infinite loops on layers which loopback, this
+		// cannot be prevented since layer IDs are not merkle hashes
+		if layersSeen[layer.ID] {
```

Comment:
Should this be an error condition? It seems like we'd still have to add such layers to the manifest to ensure it is correctly formed, even if they are repeats.